### PR TITLE
Rich Text Parsing

### PR DESCRIPTION
**Pull request by chirag-ghosh**

### CODE REVIEW

**Review by harshkhandeparkar — APPROVED** [review]
:orz:. It also works for me locally.

File: digester/main.go [hunk 1]
```diff
@@ -239,12 +246,26 @@ func main() {
...
+					text = "<em>Archived the channel</em>"
+				} else if message.SubType == "channel_leave" {
+					text = "<em>Left the channel</em>"
+				} else if len(message.Blocks) > 0 {
+					text = parseMessage(message.Blocks)
+				}
+				if text == "" {
+					text = message.Text
+				}
+
```
Comment (rajivharlalka) [inline digester/main.go:262]: I guess a switch case would look more elegant here, maybe

File: digester/parser.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	}
+	if element.Style.TextStyle.Italic {
+		result = "*" + result + "*"
+	}
+	if element.Style.TextStyle.Strike {
+		result = "~~" + result + "~~"
+	}
+	if element.Style.TextStyle.Code {
+		result = "`" + result + "`"
+	}
```
Comment (rajivharlalka) [inline digester/parser.go:113]: would be nice to have this in a function, again more elegant

File: digester/parser.go [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+	for i := 0; i < leadingSpacesCount; i++ {
+		result = " " + result
+	}
```
Comment (rajivharlalka) [inline digester/parser.go:117]: `result = strings.Repeat(" ", leadingSpacesCount) + result`


reduces complexity from n^2 to n.

File: digester/parser.go (hunk 2 above)
Comment (chirag-ghosh) [reply digester/parser.go:113]: did you mean Switch by any chance?

Comment (rajivharlalka) [reply digester/parser.go:113]: umm no. meant a function itself here. though becoming a boilerplate and quite redundant code, having simple functions to deal such stuff helps, though only if it is being reused elsewhere. If this is a one time thing, feel free to ignore the comment and resolve it.

File: digester/main.go (hunk 1 above)
Comment (chirag-ghosh) [reply digester/main.go:262]: Handled in #38

File: digester/parser.go (hunk 3 above)
Comment (chirag-ghosh) [reply digester/parser.go:117]: Handled in #38

**Pull request merged**